### PR TITLE
tail plugin: Implement the "GaugePersist" type.

### PR DESCRIPTION
**Pull request by octo**

### CODE REVIEW

File: src/daemon/utils_match.c [hunk 1]
```diff
@@ -295,14 +295,13 @@ cu_match_t *match_create_simple (const char *regex,
 
 void match_value_reset (cu_match_value_t *mv)
 {
-  if (mv == NULL)
+  if ((mv == NULL)
+      || ((mv->ds_type & UTILS_MATCH_DS_TYPE_GAUGE) == 0)
+      || ((mv->ds_type & UTILS_MATCH_CF_GAUGE_PERSIST) != 0))
```
Comment (rpv-tomsk) [inline src/daemon/utils_match.c]: I think what this piece of logic is hard to read.
It is unclean what this really is:

```
if (mv == NULL)
  return;

if (mv->ds_type & UTILS_MATCH_DS_TYPE_GAUGE)
 {
  if (mv->ds_type & UTILS_MATCH_CF_GAUGE_PERSIST)
    return;

   mv->value.gauge = NAN;
   mv->values_num = 0;
 }
```

File: src/collectd.conf.pod [hunk 2]
```diff
@@ -7144,6 +7144,13 @@ Use the greatest number only.
 
 Use the last number found.
 
+=item B<GaugePersist>
+
+Use the last number found. The number if not reset at the end of an interval.
```
Comment (rpv-tomsk) [inline src/collectd.conf.pod]: ...**is** not reset...

**Comment by octo** [comment]
@rpv-tomsk how's this?

File: src/daemon/utils_match.c (hunk 1 above)
Comment (octo) [inline src/daemon/utils_match.c]: Fixed, thanks!

**Comment by rpv-tomsk** [comment]
Hi! I think it much better currently :+1:

**Pull request merged**